samples/cpp/tutorial_code/calib3d/real_time_pose_estimation/src/main_registration.cpp

```diff
@@ -58,211 +46,250 @@ Model model;
+
+            registration.registerPoint(point_2d, point_3d);
+            if( registration.getNumRegist() == registration.getNumMax() ) end_registration = true;
+        }
+    }
 }
 
 /**  Main program  **/
-int main()
+int main(int argc, char *argv[])
 {
-
-  help();
-
-  // load a mesh given the *.ply file path
-  mesh.load(ply_read_path);
-
-  // set parameters
-  int numKeyPoints = 10000;
-
-  //Instantiate robust matcher: detector, extractor, matcher
-  RobustMatcher rmatcher;
-  Ptr<FeatureDetector> detector = ORB::create(numKeyPoints);
-  rmatcher.setFeatureDetector(detector);
-
-  /**  GROUND TRUTH OF THE FIRST IMAGE  **/
-
-  // Create & Open Window
-  namedWindow("MODEL REGISTRATION", WINDOW_KEEPRATIO);
-
-  // Set up the mouse events
-  setMouseCallback("MODEL REGISTRATION", onMouseModelRegistration, 0 );
-
-  // Open the image to register
-  Mat img_in = imread(img_path, IMREAD_COLOR);
-  Mat img_vis = img_in.clone();
-
-  if (!img_in.data) {
-    cout << "Could not open or find the image" << endl;
-    return -1;
-  }
-
-  // Set the number of points to register
-  int num_registrations = n;
-  registration.setNumMax(num_registrations);
-
-  cout << "Click the box corners ..." << endl;
-  cout << "Waiting ..." << endl;
-
-  // Loop until all the points are registered
-  while ( waitKey(30) < 0 )
-  {
-    // Refresh debug image
-    img_vis = img_in.clone();
-
-    // Current registered points
-    vector<Point2f> list_points2d = registration.get_points2d();
-    vector<Point3f> list_points3d = registration.get_points3d();
-
-    // Draw current registered points
-    drawPoints(img_vis, list_points2d, list_points3d, red);
-
-    // If the registration is not finished, draw which 3D point we have to register.
-    // If the registration is finished, breaks the loop.
-    if (!end_registration)
+    help();
+
+    const String keys =
+            "{help h        |      | print this message                                                 }"
+            "{image i       |      | path to input image                                                }"
+            "{model         |      | path to output yml model                                           }"
+            "{mesh          |      | path to ply mesh                                                   }"
+            "{keypoints k   |2000  | number of keypoints to detect (only for ORB)                       }"
+            "{feature       |ORB   | feature name (ORB, KAZE, AKAZE, BRISK, SIFT, SURF, BINBOOST, VGG)  }"
+            ;
+    CommandLineParser parser(argc, argv, keys);
+
+    string tutorial_path = "../../samples/cpp/tutorial_code/calib3d/real_time_pose_estimation/"; // path to tutorial
+
+    string img_path = tutorial_path + "Data/resized_IMG_3875.JPG";  // image to register
+    string ply_read_path = tutorial_path + "Data/box.ply";          // object mesh
+    string write_path = tutorial_path + "Data/cookies_ORB.yml";     // output file
```

Comment:
Could you try to use `samples::findFile()` from here: https://github.com/opencv/opencv/pull/12354 ?
Perhaps "../../samples" prefix should be dropped.

Reply:
This works with the updated changes (on Ubuntu).